network/p2p/test/unicast_manager_fixture.go

```diff
@@ -0,0 +1,69 @@
+package p2ptest
+
+import (
+	"context"
+	"time"
+
+	libp2pnet "github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/multiformats/go-multiaddr"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/network/p2p"
+	"github.com/onflow/flow-go/network/p2p/unicast"
+	"github.com/onflow/flow-go/network/p2p/unicast/stream"
+)
+
+// UnicastManagerFixture unicast manager fixture that can be used to override the default unicast manager for libp2p nodes.
+type UnicastManagerFixture struct {
+	*unicast.Manager
+}
+
+// UnicastManagerFixtureFactory returns a new UnicastManagerFixture.
+func UnicastManagerFixtureFactory() p2p.UnicastManagerFactoryFunc {
```

Comment:
To reduce the code complexity, please consider integrating this feature through a config on the `unicast.Manager`, i.e., setting the number of concurrent outbound connections as a parameter.

Reply:
removed obsolete fixture and factory https://github.com/onflow/flow-go/pull/4022/commits/b7c8d710aedf95e497647f51eded0f826dfc324b